Add custom sudo command support

afontaine opened the pull request:
This is useful for nodes that utilize `doas` instead of `sudo`.

---

This is my first rust anything, so I'm not sure if any of this is correct (or if I can write unit tests).

I need to add documentation still as well, but getting it up in case my rust is bad was my first priority.

Closes #145

afontaine commented on src/lib.rs:
```diff
@@ -392,6 +393,14 @@ impl<'a> DeployData<'a> {
         Ok(profile_user)
     }
+
+    fn get_sudo(&'a self) -> Result<String, DeployDataDefsError> {
+       let sudo = match self.merged_settings.sudo {
+           Some(ref x) => x.clone(),
+           None => "sudo -u".to_string()
+        };
+        Ok(sudo)
+    }
```
Does it make sense to provide a default value here, or are defaults hidden somewhere else?

balsoft replied:
This looks fine to me :)

afontaine commented on src/lib.rs:
```diff
@@ -392,6 +393,14 @@ impl<'a> DeployData<'a> {
         Ok(profile_user)
     }
+
+    fn get_sudo(&'a self) -> Result<String, DeployDataDefsError> {
```
Does this need to be a `Result` type? I don't think there can be any errors to return here :thinking:

balsoft replied:
No, it doesn't. And if it had to be a `Result`, the `Error` type should have been different anyways :)

balsoft commented:
This is missing some changes:
```
diff --git a/src/cli.rs b/src/cli.rs
index 61890e4..cc5a3ac 100644
--- a/src/cli.rs
+++ b/src/cli.rs
@@ -89,6 +89,9 @@ pub struct Opts {
     /// Revoke all previously succeeded deploys when deploying multiple profiles
     #[clap(long)]
     rollback_succeeded: Option<bool>,
+    /// Which sudo command to use. Must accept at least two arguments: user name to execute commands as and the rest is the command to execute
+    #[clap(long)]
+    sudo: Option<String>,
 }
 
 /// Returns if the available Nix installation supports flakes
@@ -635,6 +638,7 @@ pub async fn run(args: Option<&ArgMatches>) -> Result<(), RunError> {
         temp_path: opts.temp_path,
         confirm_timeout: opts.confirm_timeout,
         dry_activate: opts.dry_activate,
+        sudo: opts.sudo,
     };
 
     let supports_flakes = test_flake_support().await.map_err(RunError::FlakeTest)?;
```

afontaine commented:
Awesome, thank you @balsoft!

I've added the CLI option (correct?), as well as un-`Result`ed the fetch command
and added a blurb in the README. Does this all look good?
